sphinx-mode.el

```diff
@@ -122,10 +122,54 @@ If BUFFER is not given use the `current-buffer'."
     (find-file (plist-get target :file))
     (goto-char (plist-get target :point))))
 
+(defun sphinx-compile ()
+  "Runs 'make' in project root directory, prompting user for target format. Run 'make help' to see list of target formats."
+  (interactive)
+    (let
+      (
+       (buffer-name-base (file-name-base (buffer-file-name)))
+       (project-root-dir (locate-dominating-file buffer-file-name "Makefile"))
+       (target-format (read-from-minibuffer "Make (default html): " nil nil nil nil "html" nil))
+       )
+      (let
+	  (
+	   (file-rel-path (file-relative-name buffer-file-name project-root-dir))
+	   )
+        (shell-command (concat "make -C " project-root-dir " " target-format)))))
+
+(defun sphinx-compile-and-view ()
+  "Runs sphinx-compile and then views compiled version of current source file."
+  (interactive)
+    (let
+      (
+       (buffer-name-base (file-name-base (buffer-file-name)))
+       (project-root-dir (locate-dominating-file buffer-file-name "Makefile"))
+       )
+      (let
+	  (
+	   (file-rel-path (file-relative-name buffer-file-name project-root-dir))
+	   )
+        (shell-command (concat "make -C " project-root-dir " html"))
+  (interactive)
```

Comment:
It seems you duplicated this `(interactive)` form here probaby by accident when refactoring the code.